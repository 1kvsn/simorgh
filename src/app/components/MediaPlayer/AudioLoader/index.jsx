import React from 'react';
import { node } from 'prop-types';
import styled from '@emotion/styled';
import { keyframes } from '@emotion/react';

const Wrapper = styled.div`
  position: relative;
`;

const Overlay = styled.div`
  .no-js & {
    display: none;
  }
  display: flex;
  justify-content: center;
  align-items: center;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
`;

const spin = keyframes`
0% {
  transform: rotate(0deg);
}
100% {
  transform: rotate(360deg);
}
`;

const Spinner = styled.div`
  width: 5rem;
  height: 5rem;
  border-radius: 50%;
  border: 0.3rem solid rgba(0, 0, 0, 0.2);
  border-left-color: rgba(0, 0, 0, 0.8);
  animation: ${spin} 1.1s infinite linear;
`;

const AudioLoader = ({ children }) => {
  return (
    <Wrapper>
      <Overlay data-testid="audio-loader-overlay" aria-hidden="true">
        <Spinner />
      </Overlay>
      {children}
    </Wrapper>
  );
};

AudioLoader.propTypes = {
  children: node.isRequired,
};

export default AudioLoader;
