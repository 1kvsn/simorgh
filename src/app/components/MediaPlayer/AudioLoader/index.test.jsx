import React from 'react';
import { render } from '@testing-library/react';
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import '@testing-library/jest-dom/extend-expect';

import AudioLoader from '.';

describe('Audio Loader', () => {
  shouldMatchSnapshot(
    'should match snapshot',
    <AudioLoader>Content</AudioLoader>,
  );

  it('should transclude the children', () => {
    const { getByText } = render(<AudioLoader>Test Content</AudioLoader>);

    expect(getByText('Test Content')).toBeInTheDocument();
  });

  it('should render an aria-hidden overlay', () => {
    const { container } = render(<AudioLoader>Test Content</AudioLoader>);

    const overlay = container.querySelector(
      '[data-testid=audio-loader-overlay]',
    );
    expect(overlay).toBeInTheDocument();
    expect(overlay.getAttribute('aria-hidden')).toEqual('true');
  });
});
