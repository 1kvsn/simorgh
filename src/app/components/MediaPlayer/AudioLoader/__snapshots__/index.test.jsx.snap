// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Audio Loader should match snapshot 1`] = `
@keyframes animation-0 {
  0% {
    -webkit-transform: rotate(0deg);
    -moz-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  100% {
    -webkit-transform: rotate(360deg);
    -moz-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

.emotion-0 {
  position: relative;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
}

.emotion-4 {
  width: 5rem;
  height: 5rem;
  border-radius: 50%;
  border: 0.3rem solid rgba(0, 0, 0, 0.2);
  border-left-color: rgba(0, 0, 0, 0.8);
  -webkit-animation: animation-0 1.1s infinite linear;
  animation: animation-0 1.1s infinite linear;
}

<div>
  <div
    class="emotion-0 emotion-1"
  >
    <div
      aria-hidden="true"
      class="emotion-2 emotion-3"
      data-testid="audio-loader-overlay"
    >
      <div
        class="emotion-4 emotion-5"
      />
    </div>
    Content
  </div>
</div>
`;
